### PR TITLE
enh: start refactoring runAgent

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

Some ground work for https://github.com/dust-tt/dust/issues/4602

This is the first step if changing the structure of `runAgent` for the mutli-actions future.
I just extracted the "pickAction" and "runAction" parts of the logic in separate functions.

Next steps:
- move inputs generation to pickAction (closer to future setup)
- make `generation` an actual action (so change the action config type to also union on generation)
- make `pickAction` rely on `forceUseAtIteration` (and error if multi actions for now)

## Risk

This particular PR is a total no-op, just lightweight refactoring

### CODE REVIEW

File: front/lib/api/assistant/actions/dust_app_run.ts [hunk 1]
```diff
@@ -222,13 +222,19 @@ export async function renderDustAppRunActionByModelId(
...
 // any output associated. The error is expected to be stored by the caller on the parent agent
 // message.
-export async function* runDustApp(
-  auth: Authenticator,
-  configuration: AgentConfigurationType,
-  conversation: ConversationType,
-  userMessage: UserMessageType,
-  agentMessage: AgentMessageType
-): AsyncGenerator<
+export async function* runDustApp({
```
Comment (flvndvd) [inline front/lib/api/assistant/actions/dust_app_run.ts:225]: Could we keep `auth` as a standalone parameter on all those functions?

File: front/lib/api/assistant/agent.ts [hunk 2]
```diff
@@ -409,3 +272,184 @@ export async function* runAgent(
...
+  void auth;
+  void conversation;
+
+  const action = deprecatedGetFirstActionConfiguration(configuration);
+
+  if (action !== null) {
+    return action;
+  }
+
+  return null;
```
Comment (flvndvd) [inline front/lib/api/assistant/agent.ts:293]:
```suggestion
  return action;
```

File: front/lib/api/assistant/agent.ts [hunk 3]
```diff
@@ -409,3 +272,184 @@ export async function* runAgent(
...
+  configuration: AgentConfigurationType,
+  conversation: ConversationType,
+  userMessage: UserMessageType,
+  agentMessage: AgentMessageType,
+  action: AgentActionConfigurationType
+): AsyncGenerator<
+  AgentErrorEvent | AgentActionEvent | AgentActionSuccessEvent,
+  void
+> {
+  if (isRetrievalConfiguration(action)) {
```
Comment (flvndvd) [inline front/lib/api/assistant/agent.ts:307]: Can we create one function per action type?

File: front/lib/api/assistant/agent.ts (hunk 2 above)
Comment (fontanierh) [reply front/lib/api/assistant/agent.ts:293]: 😅

File: front/lib/api/assistant/agent.ts (hunk 3 above)
Comment (fontanierh) [reply front/lib/api/assistant/agent.ts:307]: Yes, but I'd argue this should rather be done when we refactor actions into classes (which is planned). We likely want `action.run()`

File: front/lib/api/assistant/actions/dust_app_run.ts (hunk 1 above)
Comment (fontanierh) [reply front/lib/api/assistant/actions/dust_app_run.ts]: Done

**Review by flvndvd — APPROVED** [review]

**Pull request merged**